packages/server/src/utils/booking.ts

```diff
@@ -0,0 +1,142 @@
+import { validateAccountId } from "./account";
+import { insertNewBooking, BookingStatus } from "../db/bookings";
+
+interface NewBookingData {
+  accountId: string;
+  arrivalTime: string;
+  departureTime: string;
+  comments: string;
+  companions: string[];
+}
+
+interface VerifiedNewBookingData {
+  accountId: string;
+  arrivalTime: Date;
+  departureTime: Date;
+  comments: string;
+  companions: string[];
+}
+export const createNewBooking = async (
+  newBookingData: VerifiedNewBookingData,
+) => {
+  const formattedBooking = {
+    booker_id: newBookingData.accountId,
+    arrival_time: newBookingData.arrivalTime,
+    departure_time: newBookingData.departureTime,
+    comments: newBookingData.comments,
+    companions: newBookingData.companions,
+    booking_status: "pending" as BookingStatus,
+  };
+
+  await insertNewBooking(formattedBooking);
+};
+
+export const validateNewBookingData = async (
```

Comment:
Why not using something lie yup to validate ?

Reply:
So I didn't know about yup but yeah I thought about using joi too, just thought we might want to add more validation logic than just types, like checking the account exists in our db etc so I created a function :))